src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java

```diff
@@ -199,13 +200,41 @@ public void visitToken(DetailAST ast) {
             case TokenTypes.PARAMETER_DEF:
                 processVariable(ast);
                 break;
-
+            case TokenTypes.LAMBDA:
+                processLambda(ast);
+                break;
             default:
                 visitOtherTokens(ast, type);
         }
     }
 
     /**
+     * Process a lambda token.
+     * Checks whether a lambda parameter shadows a field.
+     * @param ast the lambda token.
+     */
+    private void processLambda(DetailAST ast) {
+        final DetailAST firstChild = ast.getFirstChild();
+        if (firstChild.getType() == TokenTypes.IDENT) {
+            // Parameter type of lambda expression is omitted.
+            // ANTLR parces parameter of lambda expression
+            // which type is omitted as an identifier.
+            // Does it shadow a field?
```

Comment:
1)
> `// Does it shadow a field?`

Is this a comment to line `final String name = firstChild.getText();`?

2) Maybe we can extend Javadoc instead of putting so long comments in method body?
3) `parces` -> `parses`